scripts/lighthouse/README.md

```diff
@@ -0,0 +1,21 @@
+# Lighthouse
+
+The budget is set as a single 'score', from `0` to `1`, which is calculated as a weighted average of several different audits that Lighthouse performs.
+
+We currently use Lighthouse 5, and can see details on the weighted averages it uses in the [Lighthouse 5 performance docs](https://web.dev/performance-scoring/#lighthouse-5).
+
+## Running Lighthouse locally
+
+Be aware Lighthouse results will be better on your local machine versus other sources as your laptop is more powerful. 
+
+To run locally first start up the docs site locally
+
+```js
+npm start
```

Comment:
If that is the case then we should probably make a separate project called backpack-lighthouse and keep the scripts there to run against the live site - or even just use https://developers.google.com/speed/pagespeed/insights/ periodically? wdyt?

Reply:
I agree with Hugo, a lot of the assets optimisations and cache will not happen in the dev server, so it's not a realistic scenario.

Reply:
What happens if we run the docs site as a production build would that simulate the same?

Reply:
We can run the dist build and start a webserver in the resulting folder.

`npm run docs:dist && cd dist && python -m http.server` roughly. We might want to use a node base server instead of `python` but the idea is the same

Reply:
Not exactly, ideally we should test if in prods our artefacts are cached and compressed correctly, when applicable, and that will only be a realist test if we use the same environment we use in prod, which is not possible locally because we use `github.io` to do that.

If we do run locally it will be better to use the prod build and spin up a python server here for example.

Reply:
Ok so I see three things here:

1. We run as Hugo mentioned above for this I am using `npm run docs:dist && cd dist && npx http-server` which will build it as a prod build and then run it on a node server
2. The scripts are moved out of here to a separate repo and they are run against backpack.github.io as they would have no reason to live in here if we aren't running it on the code - also we would have to see how github handles multiple requests from automated software as I would imagine that github pages would put somethiing in the way of this
3. We don't do these scripts at all and as mentioned above use a prebuilt tool and manually feed URLs in that we are concerned about

Reply:
We can configure a local server to be similar in caching etc to GitHub pages?

Reply:
I don't know what caching GH Pages uses so would probably never accurate

Reply:
I guess one thing to note is right now even locally without the full caching, etc. The docs fails on performance on desktop, so even if we replicate it - the results could be worse or maybe even better but I think the idea with this PR is to give us a rough guide on the issues for us to investigate fixes

Reply:
We can find out by looking at the cache headers

Reply:
`max-age` is 600 for everything

Reply:
So from a quick look this is what it does that is important.
![Screenshot 2020-04-21 at 14 55 21](https://user-images.githubusercontent.com/1457263/79874422-4282c200-83e0-11ea-9f3b-51d6da2a9004.png)

Etag and gzip are not that simple to replicate locally, also we should stick with the python server if we do it, express servers are know for being really bad and handling static files.

Reply:
But considering all of this I'd go for running against `github.io` and not locally.

Reply:
@tiagoengel - I would disagree and say ideally we need to do it on both as we need to make sure changes we make to the docs site don't impact performance before it hits production as that defeats the purpose of making sure its performant you shouldn't be shipping the issues to people then deciding to fix afterwards. 

I think we need to agree/come up with a way to have both so that we can ensure we are delivering consistent value during development and live

Reply:
> Etag and gzip are not that simple to replicate locally, also we should stick with the python server if we do it, express servers are known for being really bad and handling static files.

So the `http-server` package we can setup gzip optimisations and cache control at least but don't think etag is supported.

This lib also is not running an express server in node but a plain and simple http server - more details can be found here: https://www.npmjs.com/package/http-server

Reply:
Another idea that just came to mind, we can use some good production ready docker image and run the docs site inside it for the test, that would be more close to prod environment.

Reply:
My issue with running locally in a different environment is that we will trigger errors that are not a problem in prod, and vice versa, which for me defeats a bit the purpose of it. Plus I don't think this is such a critical thing that fixing the problem after is being deploys is such a big issue.

Reply:
On the different environment side of things I also did make a note of there maybe differences in the results here: https://github.com/Skyscanner/backpack-docs/blob/BPK-2590-run-docs-through-lighthouse/scripts/lighthouse/README.md#running-lighthouse-locally

This is how one of the internal projects do their lighthouse testing which is where this all took inspro from 😄 

But I think after discussion we have had internally we have an agreement on what the outcome of this should be

Reply:
I agree with @olliecurtis the value of this is from it being pre-merge